pytorch_lightning/plugins/training_type/dp.py

```diff
@@ -52,30 +52,24 @@ def setup(self, model):
         model.to(self.root_device)
         self._model = DataParallel(LightningParallelModule(model), self.parallel_devices)
 
-    def reduce(self, tensor, *args, **kwargs):
+    def reduce(self, collection: _METRIC_COLLECTION, *args, **kwargs) -> _METRIC_COLLECTION:
```

Comment:
```suggestion
    def reduce(self, collection: _METRIC_COLLECTION, *_, **__) -> _METRIC_COLLECTION:
```

Reply:
please keep it, for consistency with superclass and docstring

Reply:
yes, otherwise mypy may also fail